### ISSUE TITLE
Auth redirect URL receiving extra query params

### ISSUE THREAD

**Issue by justinh-kairos** [strapi/strapi#13752]
### Required System information
- Node.js version: 14.18.3
- NPM version: 6.14.15
- Strapi version: 4.2.2
- Database: MySQL
- Operating system: Linux (via Azure App Service)

### Describe the bug

When using Microsoft auth provider, attempting to log in is returning the `access_key` twice to the redirect URL. The redirect URL is formatted like this (decoded for ease of reading):
`https://{redirect_base_url}/redirect?access_token={token}&raw[token_type]=Bearer&raw[scope]=openid profile User.Read email&raw[expires_in]=4088&raw[ext_expires_in]=4088&raw[access_token]={token}`

As you can see, the token is sent twice, in addition with a bunch of extra `raw[]` information. Is there a way to disable this? It's causing issues with my web app (instead of hitting my `/redirect` endpoint, I'm getting a 404 error that I haven't been able to track down the reason for).

NOTE: Removing all of the `raw[]` query params resolves the issue, which then correctly routes to my `/redirect `endpoint and successfully logs in the user.

### Expected behavior

Either don't send the `raw[]` query params, or allow configuration to remove them from the login callback

**Comment by derrickmehaffy** [comment]
I -think- this is grant/purest and not us 🤔

**Comment by justinh-kairos** [comment]
The strangest part is, I *swear* this was working previously, and as far as I remember, the only change I made was updating strapi from 4.2.0 to 4.2.2.

**Comment by derrickmehaffy** [comment]
I don't believe this would break anything but the only PR that could have is https://github.com/strapi/strapi/pull/13435

**Comment by justinh-kairos** [comment]
As a side note: the 404 error I'm seeing is because the URL is too long.

**Comment by justinh-kairos** [comment]
Correct, but the fact that the access token is being included in the URL twice is what's bloating the URL length.

**Comment by derrickmehaffy** [comment]
@justinh-kairos I can't find any references in our code that would cause the RAW params to be included meaning it either comes from grant/purest (the lib we use for 3rd party auth in U&P) of which I can't find a reference their either or it comes from Microsoft which we have no control over.

**Comment by justinh-kairos** [comment]
I'm not sure how it could be coming from Microsoft. The redirect chain during login looks like this:

- {strapi_url}/api/connect/microsoft
- https://login.microsoftonline.com/.../authorize?client_id={...}&response_type=code&redirect_uri={...}&scope=user.read
- {strapi_url}/api/connect/microsoft/callback?code={...}&session_state={...}
- {my_callback_url}?access_token={...}&raw[token_type]=Bearer...etc...

It definitely seems like those parameters are being tacked on between the strapi callback and my app's callback. If it's coming from strapi, or grant, or purest, it doesn't really make a difference to me unless I have a way to configure/prevent it. If I don't, then I view it as a bug with the U&P plugin.

**Comment by derrickmehaffy** [comment]
That would make sense, are you up for submitting a PR?

**Comment by PaulCorbeau** [comment]
I'm struggling for days also with Microsoft Auth and i'm not able to understand what is wrong.
I'm using Microsoft SSO.
